eng/mgmt/automation/generate_data.py

```diff
@@ -47,16 +47,40 @@ def sdk_automation_typespec(config: dict) -> List[dict]:
         service = None
         module = None
         try:
-            cmd = ['pwsh', './eng/common/scripts/TypeSpec-Project-Process.ps1', tsp_dir, head_sha, repo_url]
-            logging.info('Command line: ' + ' '.join(cmd))
-            output = subprocess.check_output(cmd, cwd=sdk_root)
-            output_str = str(output, 'utf-8')
-            script_return = output_str.splitlines()[-1] # the path to sdk folder
-            sdk_folder = os.path.relpath(script_return, sdk_root)
+            def remove_prefix(text, prefix):
+                if text.startswith(prefix):
+                    return text[len(prefix):]
+                return text
+
+            def find_sdk_folder():
+                cmd = ['git', 'add', '.']
+                check_call(cmd, sdk_root)
+
+                cmd = ['git', 'status', '--porcelain', '**/tsp-location.yaml']
+                logging.info('Command line: ' + ' '.join(cmd))
+                output = subprocess.check_output(cmd, cwd=sdk_root)
```

Comment:
nit, a new function `check_output`? Similar to `check_call`.https://github.com/Azure/azure-sdk-for-java/blob/9531269a2e6522cb348ae79b1f498fffbe0bb23f/eng/mgmt/automation/generate_data.py#L128-L130

Can we all use `check_output`?

Reply:
We currently only have this twice. So maybe factor this out when we write the 3rd `check_output` in code.